### PR TITLE
Add stylesheet to fileinput button

### LINKED ISSUE #6475: Make FileInput Button adhere to the design
``` python
import panel as pn

pn.extension(design="bootstrap")

file_input_stylesheet = """
input[type="file"].bk-input {
  padding-left: 10px;
}
input[type="file"]::file-selector-button {
  background: var(--surface-color);
  color: var(--surface-text-color);
}
"""

pn.widgets.FileInput().servable()
pn.widgets.Button(name="Button").servable()
pn.widgets.FileInput(stylesheets=[file_input_stylesheet]).servable()
```

image

### PR DESCRIPTION
**Pull request by Hoxbro**
Resolves #6475

I don't know if this is the best place to put the CSS in. 

image

Code in the original issue.

### CODE REVIEW

File: panel/dist/css/button.css [hunk 1]
```diff
@@ -157,3 +157,12 @@
 :host(.bk-panel-models-widgets-FileDownload) .bk-btn .bk-TablerIcon {
   margin-left: 0.5em;
 }
+
+input[type='file'].bk-input {
+  padding-left: 10px;
+}
+
+input[type='file']::file-selector-button {
```
Comment (philippjfr) [inline panel/dist/css/button.css:165]: How does this selector work?

Comment (Hoxbro) [reply panel/dist/css/button.css:165]: I found it by googling, and it seemed to do the trick :upside_down_face:  

https://developer.mozilla.org/en-US/docs/Web/CSS/::file-selector-button

Comment (philippjfr) [reply panel/dist/css/button.css:165]: Ah, makes sense!

**Review by philippjfr — APPROVED** [review]

**Comment by monodera** [comment]
This change seems to affect the other templates as well and to me these does not look nice. 

Without template:
Screenshot 2024-03-28 at 14 58 32

With Material template:
Screenshot 2024-03-28 at 14 57 56

Is it possible to revert it for other templates and make the change specific to the bootstrap template (if it's not an intension)?

**Comment by Hoxbro** [comment]
I'm not sure what you dislike... Is it the height? If so, I don't think this PR changed that behavior. Try commenting out the `stylesheets` line to see if that restores the behavior.

**Comment by monodera** [comment]
I'm sorry that I forgot to show the one with Panel 1.3.8 (shown below). 

Screenshot 2024-03-28 at 22 36 23

What I don't like for the 1.4 style is the horizontal space from the left edge of the input box to the Browse button. For the Bootstrap theme, the space is equally distributed at the left, top, and bottom, but for (at least) Material and the default templates, the top and bottom paddings are zero. I feel some lack of consistency for the treatment of margins between templates.

**Pull request merged**